Objective-C/Tests/CollectionTest.m

```diff
@@ -842,13 +842,21 @@ - (void) testUseInvalidCollection: (NSString*)collectionName onAction: (void (^)
         return [col deleteIndexWithName: @"index1" error: err];
     }];
     
-    // add change listeners
-    id<CBLListenerToken> token = [col addChangeListener: ^(CBLCollectionChange *change) { }];
-    [token remove];
+    // add change listeners;
+    // to avoid stopping c4exception break point, we use ignoreException
+    [self ignoreException: ^{
+        id<CBLListenerToken> token = [col addChangeListener: ^(CBLCollectionChange *change) { }];
+        [token remove];
+    }];
+    
     
     // doc change listener
-    token = [col addDocumentChangeListenerWithID: @"doc1" listener: ^(CBLDocumentChange *change) { }];
-    [token remove];
+    // to avoid stopping c4exception break point, we use ignoreException
+    [self ignoreException: ^{
+        id<CBLListenerToken> token = [col addDocumentChangeListenerWithID: @"doc1"
+                                                                 listener: ^(CBLDocumentChange *change) { }];
+        [token remove];
+    }];
```

Comment:
These are not related to this PR. It's better to have them fixed in a separate PR to keep the PR containing only relevant stuff.